### PR TITLE
Add PHP 7.3 Compatability Checks

### PR DESCRIPTION
**Pull request by shawnhooper**
This pull request adds support for checking code for compatibility with PHP 7.3.  Note: The option to check against PHP 7.3 will only appear if the server is running on PHP 5.3 or greater.

### CODE REVIEW

**Comment by shawnhooper** [comment]
@stevenkword I've updated the changelog, version numbers, and the PHP usage stats in the readme.

**Comment by stevenkword** [comment]
> @stevenkword I've updated the changelog, version numbers, and the PHP usage stats in the readme.

Perfect thank you!  Are you able to view the travis reports below?  The failures are coming out of WPCS lint checks.  If you don't have access I can give you a copy of the logs.

**Comment by shawnhooper** [comment]
@stevenkword I was able to fix all the linting errors thrown by 'npm test'.  However, there are still a few builds that fail - see https://travis-ci.org/wpengine/phpcompat/builds/562254368 -- that I can't figure out why.

**Comment by jrfnl** [comment]
> However, there are still a few builds that fail - see https://travis-ci.org/wpengine/phpcompat/builds/562254368 -- that I can't figure out why.

That's because the Travis matrix needs updating. WP 5.2/`latest` will not run on PHP < 5.6, so the unit tests can't run on those combis.

File: composer.json [hunk 1]
```diff
@@ -7,12 +7,12 @@
 		"email": "jason@stallin.gs"
 	}],
 	"require": {
-		"squizlabs/php_codesniffer": "^2.9.1",
+		"squizlabs/php_codesniffer": "^2.9.2",
```
Comment (jrfnl) [inline composer.json]:
```suggestion
		"squizlabs/php_codesniffer": "^2.9.2 || ^3.3.1",
```

File: composer.json [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
 		"dealerdirect/phpcodesniffer-composer-installer": "^0.4.4",
-		"phpcompatibility/phpcompatibility-wp": "^1"
+		"phpcompatibility/phpcompatibility-wp": "^2"
 	},
 	"require-dev": {
-		"wp-coding-standards/wpcs": "dev-master"
+		"wp-coding-standards/wpcs": "^1"
```
Comment (jrfnl) [inline composer.json]:
```suggestion
		"wp-coding-standards/wpcs": "^1 || ^2"
```

**Comment by shawnhooper** [comment]
> > However, there are still a few builds that fail - see https://travis-ci.org/wpengine/phpcompat/builds/562254368 -- that I can't figure out why.
> 
> That's because the Travis matrix needs updating. WP 5.2/`latest` will not run on PHP < 5.6, so the unit tests can't run on those combis.

Thank you @jrfnl, I've learned something new today.  Matrix updated, and tests are now passing.

**Comment by jrfnl** [comment]
> Thank you @jrfnl, I've learned something new today. Matrix updated, and tests are now passing.

Well, I am to blame for it... :sunglasses: https://core.trac.wordpress.org/ticket/47169

**Pull request merged**